app/views/tags/_search_form.html.erb

```diff
@@ -1,12 +1,65 @@
-<% @query = {} unless @query %>
-<%= form_tag url_for(:controller => 'tags', :action => 'search'), :class => 'simple search', :role => "search", :method => :get do %>
-<fieldset class="search tags">
-  <legend>Search tags</legend>
-  <p>
-	<%= link_to_help "tag-search-text-help" %> <%= text_field_tag "query[name]", @query[:name], :id => 'tag_search', :title => ts("Enter tags")  %>
-    <%= select_tag "query[type]", options_for_select([""] + Tag::USER_DEFINED, @query[:type]), :title => ts("Choose type") %>
-    <label for="canonical" class="action">Canonical <%= check_box_tag "query[canonical]", true, @query[:canonical], :id => 'canonical' %></label>
-  <span class="submit actions"><%= submit_tag ts("Search tags"), :name => nil %></span>
-</p>
-</fieldset>
+<%= form_for @search, as: :tag_search, url: search_tags_path, html: { class: "search", method: :get } do |f| %>
+  <fieldset>
+    <dl>
+      <dt>
+        <%= f.label :name, ts("Tag name") %>
+        <%= link_to_help "tag-search-text-help" %>
+      </dt>
+      <dd>
+        <%= f.text_field :name %>
+      </dd>
+      <dt>
+        <%= f.label :fandom_names, ts("Fandoms") %>
+      </dt>
+      <dd>
+        <%= f.text_field :fandom_names, "aria-describedby" => "fandom-field-description" %>
```

Comment:
I saw your comment and I think the ARIA attribute might need to be handled like this to work with the autocomplete: https://github.com/otwcode/otwarchive/blob/e60e8226f193eed1daf13f6ab2ea45b49b58e7a9/app/views/bookmarks/_bookmark_form.html.erb#L82

Reply:
Thank you!